org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/binding/BindingsView.java

```diff
@@ -252,7 +260,17 @@ public void connectionRemoved(IConnectedEngineFacade server) {
     for (var server : SonarLintCorePlugin.getServersManager().getServers()) {
       server.addConnectedEngineListener(serverListener);
     }
-
+    
+    // add listener for when project is closed / deleted
+    AbstractProjectChangeListener.IProjectChangeHandler handler = (IProject project) -> {
+      refreshConnectionState();
+      for (var server : SonarLintCorePlugin.getServersManager().getServers()) {
+        refreshConnectionContent(server);
+      }
+    };
+    
+    projectCloseDeleteListener = new ProjectCloseDeleteListener(handler);
```

Comment:
I am not totally convinced by the need to have 2 different listeners, while in practice you have the same handler in both cases.
Also this AbstractProjectChangeListener seems a bit over-engineered to me. Why not simply having `BindingViews`  implementing IResourceChangeListener? I think this si a common practice in Eclipse codebase.